### ISSUE TITLE
Readme example doesn't compile.

### ISSUE THREAD

**Issue by StevenTCramer** [ardalis/SmartEnum#39]
dotnet new console -n SmartTest
cd SmartTest
dotnet add package Ardalis.SmartEnum --version 1.0.5
add TestEnum.cs pasted from readme.
```
public sealed class TestEnum : SmartEnum<TestEnum>
{
    public static readonly TestEnum One = new TestEnum(nameof(One), 1);
    public static readonly TestEnum Two = new TestEnum(nameof(Two), 2);
    public static readonly TestEnum Three = new TestEnum(nameof(Three), 3);

    private TestEnum(string name, int value) : base(name, value)
    {
    }
}
```

λ  dotnet build
Microsoft (R) Build Engine version 15.9.20+g88f5fadfbe for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

  Restore completed in 56.65 ms for C:\git\temp\SmartTest\SmartTest.csproj.
TestEnum.cs(5,27): error CS0305: Using the generic type 'SmartEnum<TEnum, TValue>' requires 2 type arguments [C:\git\temp\SmartTest\SmartTest.csproj]

Build FAILED.

TestEnum.cs(5,27): error CS0305: Using the generic type 'SmartEnum<TEnum, TValue>' requires 2 type arguments [C:\git\temp\SmartTest\SmartTest.csproj]
    0 Warning(s)
    1 Error(s)

----

downloaded the repo and the tests run fine.  So I am guessing what is in Nuget isn't what is in master??

[SmartTest.zip](https://github.com/ardalis/SmartEnum/files/2791405/SmartTest.zip)

**Comment by ardalis** [comment]
I got a different error than you did, which stated that the SmartEnum namespace didn't support type arguments. I added a using statement and it compiled fine:

```
using System;
using Ardalis.SmartEnum; // ADDED

namespace SmartTest
{
    class Program
    {
        static void Main(string[] args)
        {
            Console.WriteLine("Hello World!");
        }
    }

    public sealed class TestEnum : SmartEnum<TestEnum>
{
    public static readonly TestEnum One = new TestEnum(nameof(One), 1);
    public static readonly TestEnum Two = new TestEnum(nameof(Two), 2);
    public static readonly TestEnum Three = new TestEnum(nameof(Three), 3);

    private TestEnum(string name, int value) : base(name, value)
    {
    }
}
}

```

**Comment by ardalis** [comment]
I updated the README with the using statement. It was a bad decision in retrospect to have a namespace and a class `SmartEnum`...

**Comment by StevenTCramer** [comment]
Hmm I have using statement should be in the zip.   But I have it inside the namespace (Which as a rule is where I like it.)  
This fails:
```
namespace SmartTest
{
  using Ardalis.SmartEnum;

  public class TestEnum : SmartEnum<TestEnum>
  {
    public static readonly TestEnum One = new TestEnum(nameof(One), 1);
    public static readonly TestEnum Two = new TestEnum(nameof(Two), 2);
    public static readonly TestEnum Three = new TestEnum(nameof(Three), 3);

    protected TestEnum(string name, int value) : base(name, value)
    {
    }
  }
}
```
When moving it outside it compiles.  

```
using Ardalis.SmartEnum;
namespace SmartTest
{
  public class TestEnum : SmartEnum<TestEnum>
  {
    public static readonly TestEnum One = new TestEnum(nameof(One), 1);
    public static readonly TestEnum Two = new TestEnum(nameof(Two), 2);
    public static readonly TestEnum Three = new TestEnum(nameof(Three), 3);

    protected TestEnum(string name, int value) : base(name, value)
    {
    }
  }
}
```
Here is some conversation about the differences of in and out.
https://stackoverflow.com/questions/125319/should-using-directives-be-inside-or-outside-the-namespace 

But I don't readily understand why this is an issue in this case.

**Comment by ardalis** [comment]
I'm not sure why it makes a difference here, either. And to be honest I didn't open your zip, I just followed your (very helpful) command line instructions. My preference is for using directives to be the first thing in a file, and I've never run into an issue following this approach (in coding C#/.NET since it was created). I read the SO article and can see edge cases where it could be an issue but again I've never run into it that I can recall. Or if I did I fixed it by not having conflicting names, rather than by requiring putting just the right using directive in just the right place in each file (which seems rather error-prone).

**Comment by StevenTCramer** [comment]
Ok now I compile it with it outside and that too fails. https://screencast-o-matic.com/watch/cqVZop3tXZ

**Comment by ardalis** [comment]
That's not "outside" the namespace - you got rid of the namespace (to be clear).

**Comment by ardalis** [comment]
Also make sure you update to 1.0.6

**Comment by ardalis** [comment]
With 1.0.6 I can get it to work:
- Without a `namespace` wrapping the code
- With a `namespace` wrapping the code
- With or without a second type argument

**Comment by StevenTCramer** [comment]
LOL I just looked and I did get rid of the namespace... can't lie about it I sent the video... Just means I need to go to sleep.  And yes 1.0.6 it works.

Thanks.  I have been using "Smart" Enumerations for a while based on Jimmy's blog https://lostechies.com/jimmybogard/2008/08/12/enumeration-classes/

Had been tempted to role my own lib but now I don't have to.

No idea what caused this between 1.0.5 and 1.0.6
Thanks.

BTW Jimmy also puts the usings inside the namespace. see example: https://github.com/jbogard/ContosoUniversity/blob/master/src/ContosoUniversity/Features/Course/Create.cs

It often makes the usings a little shorter.  But I have never had an issue with it.  And now with 1.0.6 its fine.

Thanks for the help, sorry for the screw up on the video :) LOL.  Good night.
